src/smc-webapp/project_store.coffee

```diff
@@ -1753,6 +1824,22 @@ create_project_store_def = (name, project_id) ->
         if @public_paths?
             return immutable.fromJS(misc.copy_without(x,['id','project_id']) for _,x of @public_paths.toJS())
 
+
+    library_docs_sorted: depends('library') ->
```

Comment:
@johnjeng I hope this is what you had in mind!

Reply:
Yup, this looks great! I have no idea if the caching will have any impact, but this is what I have in mind for computed values.